From 26401df2e74e6d72e86fab165445949c9dd3eba9 Mon Sep 17 00:00:00 2001
From: Davide Beatrici <git@davidebeatrici.dev>
Date: Sat, 21 Feb 2026 02:55:40 +0100
Subject: [PATCH 1/3] archive_read_append_filter(): Keep iterating even if name
 is null

While working on the RPM format I noticed the zstd filter failed to append.

Turns out there are two unknown filters that are registered with a null name:

bidder candidate bzip2
bidder candidate compress (.Z)
bidder candidate gzip
bidder candidate lzip
bidder candidate lzma
bidder candidate xz
bidder candidate uu
bidder candidate lrzip
bidder candidate (null)
bidder candidate (null)
bidder candidate lz4
bidder candidate zstd

I'm not sure if anonymous filters are expected behavior, but I believe this change doesn't hurt.

Upstream-Status: Submitted [https://github.com/libarchive/libarchive/pull/2846]

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 libarchive/archive_read_append_filter.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libarchive/archive_read_append_filter.c b/libarchive/archive_read_append_filter.c
index cd88df11..9475b704 100644
--- a/libarchive/archive_read_append_filter.c
+++ b/libarchive/archive_read_append_filter.c
@@ -117,7 +117,7 @@ archive_read_append_filter(struct archive *_a, int code)
     bidder = a->bidders;
     for (i = 1; i < number_bidders; i++, bidder++)
     {
-      if (!bidder->name || !strcmp(bidder->name, str))
+      if (bidder->name && !strcmp(bidder->name, str))
         break;
     }
     if (!bidder->name || strcmp(bidder->name, str))
-- 
2.49.0

